Automattic/wp-calypso
# migrate `NameServers` to use react-query

#### Changes proposed in this Pull Request

* Nameservers: use React Query instead of redux state + data layer (previously reduxified in #47272)

#### Testing instructions

* Go to `/domains/manage/:site/name-servers/:domain` and try adding new name servers, editing and deleting existing ones, as well as resetting to the original nameservers and verify everything works well and as it did before.
